client/packages/common/src/ui/components/inputs/DatePickers/DateTimePickerInput/DateTimePickerInput.tsx

```diff
@@ -1,29 +1,63 @@
-import React, { FC } from 'react';
+import React, { FC, useState } from 'react';
 import { DateTimePicker, DateTimePickerProps } from '@mui/x-date-pickers';
 import { BasicTextInput } from '../../TextInput/BasicTextInput';
 import { useAppTheme } from '@common/styles';
 import { StandardTextFieldProps, TextFieldProps } from '@mui/material';
+import { DateUtils, useTranslation } from '@common/intl';
+import { getFormattedDateError } from '../BaseDatePickerInput';
 
 const TextField = (params: TextFieldProps) => {
   const textInputProps: StandardTextFieldProps = {
     ...params,
     variant: 'standard',
-    sx: { width: '150px', ...params.sx },
   };
   return <BasicTextInput {...textInputProps} />;
 };
 
 export const DateTimePickerInput: FC<
-  DateTimePickerProps<Date>
-> = (props) => {
+  Omit<DateTimePickerProps<Date>, 'onChange'> & {
+    error?: string | undefined;
+    width?: number | string;
+    label?: string;
+    onChange: (value: Date | null) => void;
+    onError?: (validationError: string, date?: Date | null) => void;
+    textFieldProps?: TextFieldProps;
+  }
+> = ({
+  error,
+  onChange,
+  onError,
+  width,
+  label,
+  textFieldProps,
+  format = 'dd/MM/yyyy HH:mm',
+  ...props
+}) => {
   const theme = useAppTheme();
+  const [internalError, setInternalError] = useState<string | null>(null);
+  const [isInitialEntry, setIsInitialEntry] = useState(true);
+  const t = useTranslation('common');
 
   return (
     <DateTimePicker
-      format="dd/MM/yyyy HH:mm"
+      format={format}
       slots={{
         textField: TextField,
       }}
+      onAccept={onChange}
+      onChange={(date, context) => {
+        const { validationError } = context;
+
+        if (validationError) {
+          const translatedError = getFormattedDateError(t, validationError);
+          if (onError) onError(translatedError, date);
+          else setInternalError(validationError ? translatedError : null);
+        }
+        if (!validationError) {
+          setIsInitialEntry(false);
+          setInternalError(null);
+        }
+      }}
```

Comment:
This differs from the standard "DatePicker" in that we don't want to run the parent "onChange" method after every change in here, as the event fires after every selection (i.e. date, then hour, then minute, etc.). So using `onAccept` instead, which only fires when the picker closes.

The downside of that is that it doesn't fire when entering values by keyboard input, so I've had to add an additional `onBlur` (in the TextField element, below) to capture this.